BlackMarket/Managers/SessionManager.swift

```diff
@@ -0,0 +1,82 @@
+//
+//  Constaints.swift
+//  swiftui-base
+//
+//  Created by Karen Stoletniy on 06/9/22.
+//
+
+import SwiftUI
+import Combine
+
+internal final class SessionManager {
+  
+  var isSessionValidPublisher: AnyPublisher<Bool, Never> {
+    currentSessionTokenPublisher
+      .filter { _ in self.publishTokenState }
```

Comment:
the operators on streams are designed to operate on the stream values. Here you could be creating a retain cycle(memory leaks). Instead, you could combine multiple publishers(i.e currentSessionTokenPublisher, shouldPublishTokenPublisher) and filter on both streams.